src/Agent.Worker/TestResults/TrxResultReader.cs

```diff
@@ -389,32 +391,135 @@ private List<TestCaseResultData> ReadActualResults(XmlNodeList resultsNodes, str
 
                 resultCreateModel.RunBy = _runUserIdRef;
 
-                if (resultCreateModel.Outcome.Equals("Failed"))
-                {
-                    XmlNode errorMessage, errorStackTrace, consoleLog, standardError;
+                List<string> resulLeveltAttachments = new List<string>() { };
```

Comment:
why you need lock in line 349?

Reply:
It is a parallel.foreach loop, that is why we need a lock